### PR TITLE
gcp backup and restore

### PR DESCRIPTION
**Pull request by leena-jawale**
Signed-off-by: Leena Jawale <ljawale@redhat.com>

### CODE REVIEW

**Comment by leena-jawale** [comment]
Backup and restore process is for release 2.2.1. Please do not merge this PR.

File: stories/topics/proc-aap-gcp-backup-and-restore.adoc
Comment (ianf77) [inline stories/topics/proc-aap-gcp-backup-and-restore.adoc:7]: Note that the restore process deploys

(not sure about deploys a new deployment is there better wording?)

SQL DB instance restored to the specified backup

File: stories/topics/proc-aap-gcp-backup-and-restore.adoc
Comment (ianf77) [inline stories/topics/proc-aap-gcp-backup-and-restore.adoc:9]: The backup process involves taking a backup of the Cloud SQL DB and filestore instances at a given point in time.
The backup playbook require an active {AAPonGCP} foundation deployment to be running.
 
The following instructions describe how to run the backup playbook.

File: stories/topics/proc-aap-gcp-backup-and-restore.adoc
Comment (ianf77) [inline stories/topics/proc-aap-gcp-backup-and-restore.adoc:11]: remove

File: stories/topics/proc-aap-gcp-backup-and-restore.adoc
Comment (ianf77) [inline stories/topics/proc-aap-gcp-backup-and-restore.adoc:13]: remove

File: stories/topics/proc-aap-gcp-backup-and-restore.adoc
Comment (ianf77) [inline stories/topics/proc-aap-gcp-backup-and-restore.adoc:15]: ll 15-17 remove

File: stories/topics/proc-aap-gcp-backup-and-restore.adoc
Comment (ianf77) [inline stories/topics/proc-aap-gcp-backup-and-restore.adoc:18]: .Procedure

File: stories/topics/proc-aap-gcp-backup-and-restore.adoc
Comment (ianf77) [inline stories/topics/proc-aap-gcp-backup-and-restore.adoc:19]: can you explain where/how they are gathered?

File: stories/topics/proc-aap-gcp-backup-and-restore.adoc
Comment (ianf77) [inline stories/topics/proc-aap-gcp-backup-and-restore.adoc:20]: for the google

File: stories/topics/proc-aap-gcp-backup-and-restore.adoc
Comment (ianf77) [inline stories/topics/proc-aap-gcp-backup-and-restore.adoc:25]: This line should go before, around line 19 to explain what you do with the input variables.

File: stories/topics/proc-aap-gcp-backup-and-restore.adoc
Comment (ianf77) [inline stories/topics/proc-aap-gcp-backup-and-restore.adoc:54]: Make a note of the

File: stories/topics/proc-aap-gcp-backup-and-restore.adoc
Comment (ianf77) [inline stories/topics/proc-aap-gcp-backup-and-restore.adoc:61]: the following resources

File: stories/topics/proc-aap-gcp-backup-and-restore.adoc
Comment (ianf77) [inline stories/topics/proc-aap-gcp-backup-and-restore.adoc:67]: process deploys a new deployment

(there it is again)

deployment, and restores the filestore and SQL DB instance to the specified backup.

File: stories/topics/proc-aap-gcp-backup-and-restore.adoc
Comment (ianf77) [inline stories/topics/proc-aap-gcp-backup-and-restore.adoc:69]: NOTE (in square brackets)
====
The restored filestore instance uses CIDR
====

File: stories/topics/proc-aap-gcp-backup-and-restore.adoc
Comment (ianf77) [inline stories/topics/proc-aap-gcp-backup-and-restore.adoc:73]: The following instructions describe how to run a playbook.

File: stories/topics/proc-aap-gcp-backup-and-restore.adoc
Comment (ianf77) [inline stories/topics/proc-aap-gcp-backup-and-restore.adoc:85]: Move this line with line 75 to explain where to do this before you start.

File: stories/topics/proc-aap-gcp-backup-and-restore.adoc
Comment (ianf77) [inline stories/topics/proc-aap-gcp-backup-and-restore.adoc:101]: How do you do this?

File: stories/topics/proc-aap-gcp-backup-and-restore.adoc
Comment (ianf77) [inline stories/topics/proc-aap-gcp-backup-and-restore.adoc:103]: See earlier for how to format a Note.

File: stories/topics/proc-aap-gcp-backup-and-restore.adoc
Comment (ianf77) [inline stories/topics/proc-aap-gcp-backup-and-restore.adoc:19]: What do you mean by 'gather'?

File: stories/topics/proc-aap-gcp-backup-and-restore.adoc
Comment (ianf77) [inline stories/topics/proc-aap-gcp-backup-and-restore.adoc:23]: And again

File: stories/topics/proc-aap-gcp-backup-and-restore.adoc
Comment (ianf77) [inline stories/topics/proc-aap-gcp-backup-and-restore.adoc:35]: What do you mean by 'Pull the docker image'? From where?

for the operational container

I don't understand 'having tag with'

File: stories/topics/proc-aap-gcp-backup-and-restore.adoc
Comment (ianf77) [inline stories/topics/proc-aap-gcp-backup-and-restore.adoc:44]: should it be 'as a container' ?

File: stories/topics/proc-aap-gcp-backup-and-restore.adoc
Comment (ianf77) [inline stories/topics/proc-aap-gcp-backup-and-restore.adoc:56]: Remove empty lines. Asciidoc will ignore them.

File: stories/topics/proc-aap-gcp-backup-and-restore.adoc
Comment (ianf77) [inline stories/topics/proc-aap-gcp-backup-and-restore.adoc:65]: Restoring the filestore and SQL database to a deployment

File: stories/topics/proc-aap-gcp-backup-and-restore.adoc
Comment (ianf77) [inline stories/topics/proc-aap-gcp-backup-and-restore.adoc:71]: Running the restored playbook

File: stories/topics/proc-aap-gcp-backup-and-restore.adoc
Comment (ianf77) [inline stories/topics/proc-aap-gcp-backup-and-restore.adoc:111]: .Running the restored playbook as a container

File: stories/topics/proc-aap-gcp-backup-and-restore.adoc
Comment (ianf77) [inline stories/topics/proc-aap-gcp-backup-and-restore.adoc:113]: path to the service account

directory where the `extra_var` file created in Step 1 of the restore process exists.

File: stories/topics/proc-aap-gcp-backup-and-restore.adoc
Comment (ianf77) [inline stories/topics/proc-aap-gcp-backup-and-restore.adoc:114]: . Run the command...

File: stories/topics/proc-aap-gcp-backup-and-restore.adoc
Comment (ianf77) [inline stories/topics/proc-aap-gcp-backup-and-restore.adoc:121]: . It can take some time for the deployment to finish and for all the containers to run.

File: stories/topics/proc-aap-gcp-backup-and-restore.adoc
Comment (ianf77) [inline stories/topics/proc-aap-gcp-backup-and-restore.adoc:123]: You can now successfully log in to {PlatformName} {ControllerName} and {HubName} using..

In addition, the job history, uploaded collections and other records are in the same state as the restored deployment.  
(this is a bit vague, what state are they in?)

**Review by ianf77 — COMMENTED** [review]
I've left some comments.

File: stories/topics/proc-aap-gcp-backup-and-restore.adoc
Comment (ianf77) [inline stories/topics/proc-aap-gcp-backup-and-restore.adoc:7]: You don't need the asterisks.

File: stories/topics/proc-aap-gcp-backup-and-restore.adoc
Comment (ianf77) [inline stories/topics/proc-aap-gcp-backup-and-restore.adoc:41]: . Replace

File: stories/topics/proc-aap-gcp-backup-and-restore.adoc
Comment (ianf77) [inline stories/topics/proc-aap-gcp-backup-and-restore.adoc:31]: See previous comment on how to format a note.

**Review by ianf77 — COMMENTED** [review]
A few more comments.

File: stories/topics/proc-aap-gcp-backup-and-restore.adoc
Comment (leena-jawale) [reply stories/topics/proc-aap-gcp-backup-and-restore.adoc:7]: want to put the statement in bold so added astericks

File: stories/topics/proc-aap-gcp-backup-and-restore.adoc
Comment (prat98) [reply stories/topics/proc-aap-gcp-backup-and-restore.adoc:30]: We will write a specific command the user would need to execute in order to do this. At the moment, we still need to figure out the exact name of the image as it hasn't been published yet. **This needs to be updated before merging!**

File: stories/topics/proc-aap-gcp-backup-and-restore.adoc
Comment (ianf77) [inline stories/topics/proc-aap-gcp-backup-and-restore.adoc:34]: The operational image tag must match the version

File: stories/topics/proc-aap-gcp-backup-and-restore.adoc
Comment (ianf77) [inline stories/topics/proc-aap-gcp-backup-and-restore.adoc:5]: To backup and restore your {PlatformNameShort} deployment,

your existing {PlatformNameShort} administration secret name

support for the {AAPonGCP} foundation deployment.

File: stories/topics/proc-aap-gcp-backup-and-restore.adoc
Comment (ianf77) [reply stories/topics/proc-aap-gcp-backup-and-restore.adoc:7]: OK

AAP -> {PlatformNameShort}

DB -> database

File: stories/topics/proc-aap-gcp-backup-and-restore.adoc
Comment (ianf77) [inline stories/topics/proc-aap-gcp-backup-and-restore.adoc:10]: DB -> database

File: stories/topics/proc-aap-gcp-backup-and-restore.adoc
Comment (ianf77) [inline stories/topics/proc-aap-gcp-backup-and-restore.adoc:66]: DB -> database

**Review by ianf77 — COMMENTED** [review]
Few more adjustments

**Review by ianf77 — APPROVED** [review]
LGTM

One note for future action.

File: stories/topics/proc-aap-gcp-backup-and-restore.adoc
Comment (leena-jawale) [reply stories/topics/proc-aap-gcp-backup-and-restore.adoc:1]: Thanks

**Comment by leena-jawale** [comment]
Hey @ianf77 , Can you please review this and #67 once again? Thanks. If looks good, we can merge both PRs

File: stories/topics/proc-aap-gcp-backup-and-restore.adoc
Comment (ianf77) [reply stories/topics/proc-aap-gcp-backup-and-restore.adoc:1]: Realistically there should be two files in the PR, this procedure and the assembly that includes it, you'll need to make the changes I mentioned to the assembly before this will work.

File: stories/topics/proc-aap-gcp-backup-and-restore.adoc
Comment (ianf77) [reply stories/topics/proc-aap-gcp-backup-and-restore.adoc:94]: Change line 94 to the above.

**Review by ianf77 — COMMENTED** [review]
A couple of comments

File: stories/topics/proc-aap-gcp-backup-and-restore.adoc
Comment (leena-jawale) [reply stories/topics/proc-aap-gcp-backup-and-restore.adoc:94]: We are creating new deployment and restoring filestore and database which were backup up to the new deployment. I think no need to say `Restoring the filestore and SQL database to a deployment` as a heading. We have description under heading which tells user 
`The restore process deploys a new deployment, and restores the filestore and SQL database instance to the specified backup.`

File: stories/topics/proc-aap-gcp-backup-and-restore.adoc
Comment (pavankrao) [inline stories/topics/proc-aap-gcp-backup-and-restore.adoc:117]: @leena-jawale : this cannot be same as any deployment that is up and running already as noted by you on slack

kindly revise the same

**Review by pavankrao — CHANGES_REQUESTED** [review]
The docs must add a `troubleshooting section` or other disclaimer with notes that tell users what they must do in the event of a restoration failure

In 2.2.1, the restore playbook is not idempotent, meaning, the user will have to delete resources associated to a failed restoration before attempting to restore once again which is critical for users to be aware of. 

The GCP web UI does not warn user about the failed restoration process. Hence, user is entitled to know about the failed restore process and delete the resources to avoid billing associated to those resources

**Comment by nathanweatherly** [comment]
GCP 2.2.1 is live-- this can be merged once Doc team approves

**Review by ianf77 — APPROVED** [review]
LGTM

File: stories/topics/proc-aap-gcp-backup-and-restore.adoc
Comment (leena-jawale) [reply stories/topics/proc-aap-gcp-backup-and-restore.adoc:100]: @pavankrao All these changes will be done in following PR

**Pull request merged**
